test/test_nn.py

```diff
@@ -11641,6 +11695,178 @@ def test_embedding_padding_idx(self, device, dtype):
+                indices_2D[bag] = torch.tensor(indices_in_bag, device=device)
+
+            return indices_2D
+
+        for mode in ['max', 'mean', 'sum']:
+            for sparse in [False]:
+                indices_1D, offsets = gen_1D_indices_offsets()
+                for padding_idx_1D in list(set(indices_1D.tolist())) + [None]:
+                    msg = f"mode: '{mode}', sparse: {sparse}, padding_idx_1D: {padding_idx_1D}"
+
+                    # If 1D input does not use a padding index, we still need one for the 2D input,
+                    # so we can add one dummy word to the weights to act as the padded word
+                    padding_idx_2D = padding_idx_1D if padding_idx_1D is not None else num_words
+                    num_words_with_padding = num_words if padding_idx_1D is not None else num_words + 1
+
+                    indices_2D = gen_2D_indices_from_1D(indices_1D, offsets, padding_idx_2D)
+
+                    weights = torch.randn(
+                        num_words_with_padding,
+                        num_features,
+                        dtype=dtype,
+                        device=device,
+                        requires_grad=True)
+                    weights_check = weights.clone().detach().requires_grad_(True)
+
+                    bag = torch.nn.functional.embedding_bag(
+                        weights,
+                        indices_1D,
+                        offsets,
+                        padding_idx=padding_idx_1D,
+                        mode=mode,
+                        sparse=sparse)
+                    bag_check = torch.nn.functional.embedding_bag(
+                        weights_check,
+                        indices_2D,
+                        padding_idx=padding_idx_2D,
+                        mode=mode,
+                        sparse=sparse)
+                    self.assertEqual(bag, bag_check, msg=msg)
+
+                    bag.sum().backward()
+                    bag_check.sum().backward()
+                    self.assertEqual(weights.grad, weights_check.grad, msg=msg)
+
+    # Check correctness of torch.nn.functional.embedding_bag forward and
+    # backward functions with padding_idx, given a 2D indices input. Compare
+    # against torch.nn.functional.embedding followed by a reduction.
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.float32, torch.float64)
+    def test_embedding_bag_2D_padding_idx(self, device, dtype):
+        # Use a Python implementation of embedding_bag with padding_idx support
+        # to check torch.nn.functional.embedding_bag correctness
+        def embedding_bag_check(indices, weights, mode, sparse, padding_idx):
+            assert padding_idx is not None
+            embedding = torch.nn.functional.embedding(
+                indices,
+                weights,
+                padding_idx=padding_idx,
+                sparse=sparse)
+
+            reduction_dim = indices.dim() - 1
+
+            if mode == 'sum' or mode == 'mean':
```

Comment:
Might be good to add a comment to this block explaining that the `padding_idx` elements are being purposefully removed before the reduction to make the comparison succeed.